oidc.go

```diff
@@ -128,7 +127,27 @@ func (a *Authenticator) callbackHandler() http.Handler {
 
 		var groups []string
 		if conf.GetStringValue("engine.groups_endpoint") != "" {
-			groups, _ = getGroups(token.AccessToken, conf.GetStringValue("engine.groups_endpoint"))
+			groupURLs := strings.Split(conf.GetStringValue("engine.groups_endpoint"), ",")
+
+			groupsOut := make(chan []string)
+			for _, groupURL := range groupURLs {
+				go func(currGroupURL string) {
+					userGroups, err := getGroups(token.AccessToken, currGroupURL)
+
+					if err != nil {
+						log.Infof("Unable to fetch groups from: %s, failed with: %s", currGroupURL, err)
+					}
+
+					groupsOut <- userGroups
+				}(groupURL)
+			}
+
+			for _ = range groupURLs {
+				select {
+				case newGroups := <-groupsOut:
+					groups = append(groups, newGroups...)
```

Comment:
may be you should check if the received `newGroups` is empty or not. not sure how append will react to `nil`. I guess testing this case will also help.

Reply:
ah, `newGroups` should be converted to the nil-equivalent of []Group, so it should work fine.

I've now added some tests to ensure that this works as intended